pyramid/config/__init__.py

```diff
@@ -903,6 +921,13 @@ def end(self):
         """
         return self.manager.pop()
 
+    def __enter__(self):
+        self.begin()
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        self.end()
```

Comment:
What do you want to signify by clearing the action state? Is it telling the user that there's a way to try stuff out and rollback?

I like the idea. Practically I think it's dangerous to claim we can do anything about rollbacks. We generally expect any exceptions at config-time to kill your app and do not document error handling for any directives. Also much of the third party extensions I've ever seen do all their work in the directive without deferring it to the commit cycle.